wdl/src/main/scala/wdl/Scatter.scala

```diff
@@ -60,7 +60,20 @@ object Scatter {
       * use however many of them it needs.
       */
     val possiblyNeededNestedOgins: Map[String, OuterGraphInputNode] = outerLookup filterNot { case (name, _) => localLookup.contains(name) } map { case (name, outerPort) =>
-      name -> OuterGraphInputNode(WomIdentifier(name), outerPort, preserveScatterIndex = false)
+      /*
+        * preserveIndexForOuterLookups indicates us whether or not nodes in the outerLookup should be considered "siblings" of the scatter in term of scatter index
+        * preserveIndexForOuterLookups = false means the outerLookup nodes are outside a scatter containing this scatter node
+        * preserveIndexForOuterLookups = true means the above predicate does not hold
+        * 
+        * When creating OGINs from those outer lookup nodes for the inner graph we want to make sure we set their preserveScatterIndex to preserveIndexForOuterLookups
+        * because they effectively represent the outer lookup nodes inside the scatter. So whether the index must be preserved depends on whether this
+        * scatter node has been asked to "preserveIndexForOuterLookups".
+        * 
+        * Note that preserveIndexForOuterLookups will always be true here as long as scatters can't be nested. Indeed the only reason for a node to be
+        * asked to NOT preserve the index is if its outerLookup is referencing nodes outside a scatter.
+        * In this case preserveIndexForOuterLookups = false would mean this scatter node is part of the inner graph of another scatter node.
+       */
+      name -> OuterGraphInputNode(WomIdentifier(name), outerPort, preserveScatterIndex = preserveIndexForOuterLookups)
```

Comment:
Discussed with faces, since this is a scatter it shouldn't be possible to nest inside another scatter so OGIN indices should always turn out to be -1, right @cjllanwarne?

Reply:
Yes, but I think this is a more general version of that (it gives us the right answer now, and also in case we add nested scatters in the future)

Reply:
@cjllanwarne @mcovarr 
So looking into this more, regardless of what we put here it won't fully work for nested scatters anyway.
From a conceptual perspective I think the "right" thing to do is to set it to `false`, even though it turns out the actual value does not matter in practice for reasons too complicated to explain here.
Happy to whiteboard this with whoever wants.